internal/pkg/earpointer/analysis.go

```diff
@@ -315,7 +350,139 @@ func (vis *visitor) visitSelect(s *ssa.Select) {
 }
 
 func (vis *visitor) visitCall(call *ssa.CallCommon, instr ssa.Instruction) {
-	// TODO: to be added.
+	if builtin, ok := call.Value.(*ssa.Builtin); ok {
+		// special handling of builtin functions
+		vis.visitBuiltin(builtin, instr)
+		return
+	}
+
+	// Collect unification constraints
+	// Here only static calls are supported.
+	// A future extension is to use a more advanced call graph.
+	paramCstrs, retCstrs :=
+		vis.collectCalleeConstraints(call, call.StaticCallee(), instr)
+	// Unify caller arguments and callee parameters with suitable context.
+	for arg, params := range paramCstrs {
+		for _, param := range params {
+			// The copy-by-value semantics is implemented by the SSA call.
+			// For example:
+			//   type S struct { ... }
+			//   var v S
+			//   g(v)
+			//   func g(a S) { ... }
+			// SSA creates a copy of v and passes it to g. Hence we directly
+			// unify this copy and g's argument "a".
+			if mayShareObject(arg) {
+				vis.unifyLocals(arg, param)
+			}
+		}
+	}
+	// Handle return value unification.
+	for callerRet, rets := range retCstrs {
+		for _, calleeRet := range rets {
+			if len(calleeRet) == 1 { // non-tuple case
+				if mayShareObject(calleeRet[0]) {
+					vis.unifyLocals(callerRet, calleeRet[0])
+				}
+			} else { // tuple case
+				// For call "return_r = f(...)", if the callee returns a tuple
+				// <r0, r1, ...>, then the i^{th} element in the tuple is
+				// connected to the caller's return_r by making
+				// "return_r[i] = r{i}".
+				for k := 0; k < len(calleeRet); k++ {
+					retV := calleeRet[k]
+					if !mayShareObject(retV) {
+						continue
+					}
+					// Implement "return_r[i] = r{i}".
+					field := Field{Name: strconv.Itoa(k)}
+					vis.processHeapAccess(retV, callerRet, field)
+				}
+			}
+		}
+	}
+}
+
+// Handle calls to builtin functions: https://golang.org/pkg/builtin/.
+func (vis *visitor) visitBuiltin(builtin *ssa.Builtin, instr ssa.Instruction) {
+	switch builtin.Name() {
+	case "append": // func append(slice []Type, elems ...Type) []Type
+		// Propagage the arguments to the return value.
+		if dst, ok := instr.(ssa.Value); ok {
+			ops := instr.Operands(nil)
+			// ops[0] is the "append" function itself.
+			for i := 1; i < len(ops); i++ {
```

Comment:
SSA does useful work here to avoid unify(l1, x), unify(l1, y), ...
For example, l1 = append(l0, x) is translated to:
  
  t1 = &t0[0:int]                            **int   // t0 is l0
  *t1 = x
  t2 = slice t0[:]                           []*int
  t3 = append(s, t2...)                []*int.  // t3 is l1

So a slice is first created and "x" is assigned to its first element, then this slice is unified with l1. Specifically, for your example, the heap is:
  {l1, l0} [0 -> x, 1 -> y]